packages/cadl-ranch-specs/http/type/union/main.tsp

```diff
@@ -3,139 +3,199 @@ import "@azure-tools/cadl-ranch-expect";
 
 using TypeSpec.Http;
 
-@doc("Test for type of union.")
+/**
+ * Describe scenarios for various combinations of unions.
+ */
 @supportedBy("dpg")
 @scenarioService("/type/union")
 namespace Type.Union;
 
-@doc("This is a base model.")
-model BaseModel {
+model Cat {
   name: string;
 }
 
-@doc("The first one of the unioned model type.")
-model Model1 extends BaseModel {
-  prop1: int32;
+model Dog {
+  bark: string;
 }
 
-@doc("The second one of the unioned model type.")
-model Model2 extends BaseModel {
-  prop2: int32;
+enum LR {
+  left,
+  right,
 }
-
-union MyNamedUnion {
-  one: Model1,
-  two: Model2,
+enum UD {
+  up,
+  down,
 }
 
-model ModelWithSimpleUnionProperty {
-  simpleUnion: int32 | int32[];
+/**
+ * Describe union of string "a" | "b" | "c"
+ */
+@route("/strings-only")
+interface StringsOnly extends GetAndSend<"a" | "b" | "c", "\"b\""> {}
+
+/**
+ * Describe union of string string | "b" | "c"
+ */
+@route("/string-extensible")
+interface StringExtensible extends GetAndSend<string | "b" | "c", "\"custom\""> {}
```

Comment:
so this could be the new extensible enum :-)

Reply:
Indeed, that’s what Laurent just filed as an issue today. If you use the named version as below you basically have an extensible enum being explicit